src/mono/mono/utils/mono-threads-mach-helper.c

```diff
@@ -122,8 +127,18 @@ mono_threads_init_dead_letter (void)
 	setObjectForKey = sel_registerName ("setObject:forKey:");
 	objectForKey = sel_registerName ("objectForKey:");
 
-	// define the dead letter class
-	mono_dead_letter_class = objc_allocateClassPair (nsobject, "MonoDeadLetter", 0);
+	
+	char *partial_class_name;
+	partial_class_name = generate_partial_unique_classname ();
+
+	char *class_name = g_strconcat ("MonoDeadLetter", partial_class_name, (const char *)NULL);
```

Comment:
You only need something that's unique in the same process, and to the current runtime. Can that be as simple as the address of the "MonoDeadLetter" constant?

```c
char *class_name = g_strdup_printf ("MonoDeadLetter%p", &"MonoDeadLetter");
```

Reply:
Crafty ;-). It works! Thanks for the suggestion.